mget.go

```diff
@@ -1,15 +1,17 @@
 package opensearchtools
 
 import (
-	"bytes"
 	"context"
 	"encoding/json"
 	"net/http"
-
-	"github.com/opensearch-project/opensearch-go/v2"
-	"github.com/opensearch-project/opensearch-go/v2/opensearchapi"
 )
 
+// MGet defines a method which knows how to make an OpenSearch multiple document get.
+// It should be implemented by a version-specific executor.
+type MGet interface {
+	MGet(ctx context.Context, req *MGetRequest) (*MGetResponse, error)
+}
+
 // MGetRequest wraps the functionality of [opensearchapi.MgetRequest] by supporting request body creation.
 // We can perform an MGetRequest as simply as:
```

Comment:
If this is the pattern we go with, this doc comment sure be updated to explicitly call out that it's a union of all OpenSearch version requests and that individual executors will listen to and ignore desired fields.

Reply:
Same for the `MGetResponse`. Maybe link and list the supported version implementations.

Reply:
@cs-marshall I updated the doc comments for this (MGetRequest) as well as MGetResponse and MGetResult. LMK what you think.